Add multi cluster set up to deployment section (#2094)
flyteorg/flyte: rsts/deployment/cluster_config/customizable_resources.rst

Adding New Customizable Resources
##################################

As a quick refresher, custom resources allow you to manage configurations for specific combinations of user projects,
domains and workflows that override default values. Examples of such resources include execution clusters, task resource

> defaults, and :std:ref:`more <flyteidl:protos/docs/admin/admin:matchableresource>`. If your goal is scalability, use :ref:`multiple clusters <multicluster-setup>`.

Reviewer: ```suggestion
defaults, and :std:ref:`more <flyteidl:protos/docs/admin/admin:matchableresource>`.
```

##################################

> For background on customizable resources, see the :ref:`deployment-cluster-config-general`.

Reviewer: Can we have this as a note after "Examples of such resources include execution clusters, task resource defaults, and more."?

> As a quick refresher, custom resources allow you to manage configurations for :ref:`specific combinations <multicluster-setup>` of user projects,

Reviewer: I'm not sure if the reference attached is right.

Let's say you want to inject a default priority annotation for your workflows. Perhaps you start off with a model where everything has a default priority but soon you realize it makes sense that workflows in your production domain should take higher priority than those in your development domain.

> Now, one of your users team require critical workflows to have a higher priority than other production workflows. Here's how you could go about building a customizable priority designation.

Reviewer: ```suggestion
Now, one of your user teams requires critical workflows to have a higher priority than other production workflows. Here's how you could go about building a customizable priority designation.
```

Flyte IDL
^^^^^^^^^

> Introduce a new :std:ref:`matchable resource <protos/docs/admin/admin:matchableresource>`, that includes a unique enum value and proto message definition. 

Reviewer: ```suggestion
Introduce a new :std:ref:`matchable resource <protos/docs/admin/admin:matchableresource>` that includes a unique enum value and proto message definition. 
```

See the changes in this `file <https://github.com/flyteorg/flyteidl/commit/b1767697705621a3fddcb332617a5304beba5bec#diff-d3c1945436aba8f7a76755d75d18e671>`__ for an example of what is required.

FlyteAdmin
^^^^^^^^^^^

> Once your IDL changes are released, update the logic of Flyte Admin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.

Reviewer: ```suggestion
Once your IDL changes are released, update the logic of FlyteAdmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```

> Flyte admin

Reviewer: ```suggestion
Flyte Admin
```

^^^^^^^^^^^

> Once your IDL changes are released, update the logic of flyteadmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.

Reviewer: ```suggestion
Once your IDL changes are released, update the logic of Flyte Admin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```

Flytekit
^^^^^^^^

> For convenience, add a FlyteCTL wrapper to update the new attributes. Refer to `this section <https://docs.flyte.org/projects/flytectl/en/latest/gen/flytectl_update_plugin-override.html>`__ for the entire set of changes required.

Reviewer: ```suggestion
For convenience, add a FlyteCTL wrapper to update the new attributes. Refer to `this PR <https://github.com/flyteorg/flytectl/pull/65>`__ for the entire set of changes required.
```

> That's it! You now have a new matchable attribute to configure according to the requirements of your users evolve.

Reviewer: ```suggestion
That's it! You now have a new matchable attribute to configure as the needs of your users evolve.
```

Flytekit
^^^^^^^^

> For convenience, add a flyte-cli wrapper for updating your new attribute. See `this PR <https://github.com/flyteorg/flytekit/pull/174>`__ for the entire set of changes required.

Reviewer: @evalsocket, do we have to replace this section with FlyteCTL? Can you help with the content?
